src/main/java/com/expense/Sulaah/service/GroupServiceImpl.java

```diff
@@ -22,57 +22,59 @@ public class GroupServiceImpl implements GroupService {
     @Autowired
     private UserRepository userRepository;
 
+	private Group group;
+
     @Override
-    public Group createGroup(int userId, Group group) {
+    public Group createGroup(int userId, String name) throws SecurityException {
         Optional<User> user = userRepository.findById(userId);
-        if (user.isPresent()) {
-            List<User> usersInGroupList = group.getUsersInGroup();
-            if (usersInGroupList == null) {
-                usersInGroupList = new ArrayList<User>();
-                group.setUsersInGroup(usersInGroupList);
-            }
-            usersInGroupList.add(user.get());
-            return groupRepository.save(group);
-        } else {
-            return null;
+        if (!user.isPresent()) {
+			throw new SecurityException("User not present with given id: " + userId);
         }
+
+		group = new Group();
+		group.setName(name);
+
+		List<User> listOfUsers = new ArrayList<>();
+		listOfUsers.add(user.get());
+		group.setUsersInGroup(listOfUsers);
+
+		return groupRepository.save(group);
     }
 
     @Override
-    public Group getGroupById(int groupId) {
-        return groupRepository.findById(groupId).orElse(null);
+    public Group getGroupById(int groupId) throws RuntimeException {
+		Optional<Group> group = groupRepository.findById(groupId);
+		if(!group.isPresent()){
+			throw new RuntimeException("Group not found with ID: " + groupId);
+		}
+
+        return group.get();
     }
 
     @Override
-    public String addMembers(int groupId, List<Integer> userIdList) {
-        Optional<Group> currentGroup = groupRepository.findById(groupId);
+    public Group addMembers(int groupId, List<Integer> userIdList) throws RuntimeException {
+        Optional<Group> groupDB = groupRepository.findById(groupId);
```

Comment:
Naming!
you changed to 'currentGroup' in last commit :)

Reply:
Resolved